### PR TITLE
[test] Added jshint and eslint

### PR DESCRIPTION
**Pull request by codydaig**
Added jshint and eslint to the gulp tasks. Eslint still needs some tweaking but is operational. They are not in the test command yet since there's a lot of issues to be resolved.

### CODE REVIEW

**Comment by Kevbyte** [comment]
LGTM

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "unused": true,
+    "strict": true,
+    "trailing": true,
+    "smarttabs": true,
+    "white": true,
+    "globals": {
+      "angular": true,
+      "io": true
+    }
+}
```
Comment (AndrewKishino) [inline .jshintrc]: newline

Comment (codydaig) [inline .jshintrc]: Fixed

**Comment by AndrewKishino** [comment]
LGTM

**Pull request merged**